src/UrlParser.elm

```diff
@@ -440,18 +442,38 @@ parseString parser pathWithQuery =
       )
 
 
+-- Same as Navigation.Location.
+--
+-- Reproduced here to avoid a dependency on that library, which does not function
+-- outside of the browser.
+
+type alias Location =
+  { href : String
+  , host : String
+  , hostname : String
+  , protocol : String
+  , origin : String
+  , port_ : String
+  , pathname : String
+  , search : String
+  , hash : String
+  , username : String
+  , password : String
+  }
+
+
 {-| Parse based on `location.pathname` and `location.search`. This parser
 ignores the hash entirely.
 -}
-parsePath : Parser (a -> a) a -> Navigation.Location -> Maybe a
+parsePath : Parser (a -> a) a -> Location -> Maybe a
```

Comment:
Causes a `MAJOR` versions bump, but does not break unit tests or the example page